### ISSUE TITLE
Tumblr / Typhoeus #url-fragment redirect loop issue with Faraday

### ISSUE THREAD

**Issue by jm3** [lostisland/faraday#566]
Can anyone confirm whether this looping redirect bug I just filed against Typhoeus is in the correct place, i.e. that it's a Typhoeus / Tumblr issue, not a Faraday issue? So many layers of :turtle:s here… :)

Typhoeus/typhoeus#512

**Comment by iMacTia** [comment]
Hi @jm3 
I think you found a legitimate bug on `FaradayMiddleware::FollowRedirects`.

The same happens if you try to do it with the `:net_http` adapter:

```ruby
conn = Faraday.new('http://whathappensinadops.tumblr.com') do |f|
  f.use FaradayMiddleware::FollowRedirects
  f.adapter :net_http
end

conn.get('/post/48847516848/when-theres-something-on-the-io-that-we-dont-even')
# FaradayMiddleware::RedirectLimitReached: too many redirects; last one to: http://happensinadops.com/post/48847516848/when-theres-something-on-the-io-that-we-dont#_=_
```

Would you mind re-opening this Issue on [faraday_middleware](https://github.com/lostisland/faraday_middleware)?

**Comment by jm3** [comment]
@iMacTia happy to file it over there. 

I'm honestly pretty proud of myself for untangling the `Ruby > Rack > Typhoeus > Farraday > Tumblr` stack at all :)

**Comment by iMacTia** [comment]
Indeed you did very well in finding the issue :) And that was just the tip of the iceberg.
Let's try to fix it on lostisland/faraday_middleware#149 now ;)
